test/Core.Test/Services/GroupServiceTests.cs

```diff
@@ -30,12 +33,116 @@ public GroupServiceTests()
+
+            await Assert.ThrowsAsync<BadRequestException>(() => _sut.SaveAsync(group));
+        }
+
+        [Fact]
+        public async Task SaveAsync_OrganizationDoesNotUseGroups_ThrowsBadRequest()
+        {
+            var organization = new Models.Table.Organization
+            {
+                Id = Guid.NewGuid(),
+            };
+            _organizationRepository
+                .GetByIdAsync(organization.Id)
+                .Returns(organization);
+            var group = new Models.Table.Group
+            {
+                OrganizationId = organization.Id,
+            };
+
+            await Assert.ThrowsAsync<BadRequestException>(() => _sut.SaveAsync(group));
+        }
+
+        [Fact]
+        public async Task DeleteUserAsync_ValidData_DeletesUserInGroupRepository()
+        {
+            // organization
+            var organization = new Models.Table.Organization
+            {
+                Id = Guid.NewGuid(),
+                UseGroups = true,
+            };
+            _organizationRepository
+                .GetByIdAsync(organization.Id)
+                .Returns(organization);
+            // user
+            var organizationUser = new Models.Table.OrganizationUser
+            {
+                Id = Guid.NewGuid(),
+                OrganizationId = organization.Id,
+            };
+            _organizationUserRepository.GetByIdAsync(organizationUser.Id)
+                .Returns(organizationUser);
+            // group
+            var group = new Models.Table.Group
+            {
+                OrganizationId = organization.Id,
+            };
+
+            await _sut.DeleteUserAsync(group, organizationUser.Id);
+
+            await _groupRepository.Received().DeleteUserAsync(group.Id, organizationUser.Id);
+        }
+
+        [Fact]
+        public async Task DeleteUserAsync_InvalidUser_ThrowsNotFound()
+        {
+            // user
+            var nonOrganizationUser = new Models.Table.OrganizationUser
+            {
+                Id = Guid.NewGuid(),
+            };
+            _organizationUserRepository.GetByIdAsync(nonOrganizationUser.Id)
+                .Returns(nonOrganizationUser);
+            // organization
+            var organizationId = Guid.NewGuid();
+            var organization = new Models.Table.Organization
+            {
+                Id = organizationId,
+            };
+            _organizationRepository
+                .GetByIdAsync(organization.Id)
+                .Returns(organization);
+            // group
+            var group = new Models.Table.Group
+            {
+                OrganizationId = organization.Id,
+            };
+
+            // user not in organization
+            await Assert.ThrowsAsync<NotFoundException>(() => _sut.DeleteUserAsync(group, nonOrganizationUser.Id));
+            // invalid user
+            await Assert.ThrowsAsync<NotFoundException>(() => _sut.DeleteUserAsync(group, Guid.NewGuid()));
         }
     }
```

Comment:
Could you add a test for `DeleteAsync` as well that just validates that both repository and event service do their thing?

Like I said, it's important that these things are married for events to be meaningful